packages/synapse-react-client/src/components/SchemaDrivenAnnotationEditor/template/ArrayFieldItemTemplate.tsx

```diff
@@ -0,0 +1,104 @@
+import React from 'react'
+import { Box, Grid } from '@mui/material'
+import {
+  ArrayFieldTemplateItemType,
+  FormContextType,
+  RJSFSchema,
+  StrictRJSFSchema,
+} from '@rjsf/utils'
+
+/** The `ArrayFieldItemTemplate` component is the template used to render an items of an array.
+ *
+ * @param props - The `ArrayFieldTemplateItemType` props for the component
+ */
+export default function ArrayFieldItemTemplate<
+  T = any,
+  S extends StrictRJSFSchema = RJSFSchema,
+  F extends FormContextType = any,
+>(props: ArrayFieldTemplateItemType<T, S, F>) {
+  const {
+    children,
+    disabled,
+    hasToolbar,
+    hasCopy,
+    hasMoveDown,
+    hasMoveUp,
+    hasRemove,
+    index,
+    onCopyIndexClick,
+    onDropIndexClick,
+    onReorderClick,
+    readonly,
+    uiSchema,
+    registry,
+    onAddIndexClick,
+  } = props
+  const { CopyButton, MoveDownButton, MoveUpButton, RemoveButton, AddButton } =
+    registry.templates.ButtonTemplates
+
+  return (
+    <Grid
+      container
+      gap={2}
+      justifyContent={'space-between'}
+      className={`array-item`}
+    >
+      <Grid item={true} xs>
+        <Box mb={2}>{children}</Box>
+      </Grid>
+      {hasToolbar && (
+        <Grid
+          item={true}
+          sx={{ alignSelf: 'start', my: 1 }}
+          display={'flex'}
+          flexDirection={'row'}
+          gap={1}
+          justifyContent={'flex-start'}
+        >
+          {(hasMoveUp || hasMoveDown) && (
+            <MoveUpButton
+              disabled={disabled || readonly || !hasMoveUp}
+              onClick={onReorderClick(index, index - 1)}
+              uiSchema={uiSchema}
+              registry={registry}
+            />
+          )}
+          {(hasMoveUp || hasMoveDown) && (
+            <MoveDownButton
+              disabled={disabled || readonly || !hasMoveDown}
+              onClick={onReorderClick(index, index + 1)}
+              uiSchema={uiSchema}
+              registry={registry}
+            />
+          )}
```

Comment:
We weren't rendering the up/down buttons for array items before, but I figure we should add them anyways, because they're very useful.